python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -324,24 +324,37 @@ def _migrate_credentials(self):
         shotgun_authentication.
         """
         sl = self.create_legacy_login_instance()
+        site, login, _ = sl._get_saved_values()
         # Call get_connection, since it will reprompt for the password if
         # for some reason it is expired now.
         connection = sl.get_connection()
-        # Extract the credentials from the old Shotgun instance and create a
-        # ShotgunUser with them. This will cache the session token as well.
-        user = ShotgunAuthenticator().create_session_user(
-            login=connection.config.user_login,
-            password=connection.config.user_password,
-            host=connection.base_url,
-            # Ugly, but this is the only way available to get at the
-            # raw http_proxy string.
-            http_proxy=sl._http_proxy
-        )
 
         # Next set the current host and user in the framework.
         dm = DefaultsManager()
-        dm.set_host(user.host)
-        dm.set_login(user.login)
+        dm.set_host(site)
```

Comment:
This was actually a bug. If we don't set the current host first, create_session_user will then cache the user inside the wrong site.